### PR TITLE
Feature login page

### PR DESCRIPTION
**Pull request by SabaLeonel**
# Add Login Form Component with Social and Email Login Options

## Description
This pull request introduces a new `LoginForm` component to the application. The login form allows users to log in using either social login providers or their email address. A seamless transition effect is included to enhance the user experience when switching between social login and email login forms. The form is styled to match the application's design standards.

## Changes
- **Added** a new `LoginForm` component in `login-form.tsx`.
- **Implemented** social login options using the `Social` component.
- **Added** a button for users to switch to the email login form.
- **Included** a smooth transition effect for toggling between social and email login forms.
- **Styled** the login form to maintain design consistency with the application.

### Social Login Form
[Include screenshot of the social login form]

### Email Login Form
{C278889B-95C9-4822-BAC4-AE76E734EB21}
{9C5F69C9-41C1-45D4-94EF-132C289A69BB}

## How to Test
1. Navigate to the login page of the application.
2. Verify that social login options (e.g., Google, Facebook) are displayed initially.
3. Click the "Continue with Email" button:
   - Ensure the email login form appears with a smooth transition effect.
   - Confirm the transition feels natural and bug-free.
4. Verify the styling of the email login form matches the application's design.
5. Test form submission for both social and email login:
   - Ensure proper handling of valid inputs.
   - Confirm validation errors for invalid inputs (e.g., incorrect email format).

## Additional Notes
- Ensure the environment variables for social login providers are correctly configured in the `.env` file:
  - Example:
    ```
    GOOGLE_CLIENT_ID=your-google-client-id
    GOOGLE_CLIENT_SECRET=your-google-client-secret
    MAILGUN_API_KEY=your-mailgun-api-key
    ```
- The email login functionality requires the Mailgun provider to be properly set up in the backend.
- If any issues occur during testing, verify the configurations of social providers and email provider services.

### CODE REVIEW

**Review by phlearning — DISMISSED** [review]

File: src/components/header.tsx
Comment (phlearning) [inline src/components/header.tsx:1]: Looks good to me

File: src/components/header.tsx [hunk 1]
```diff
@@ -15,11 +19,30 @@ export async function Header() {
...
 				<div className="flex items-center gap-4">
-					<Link href={session ? "/api/auth/signout" : "/api/auth/signin"} className="line-clamp-1 flex items-center justify-center gap-2 rounded-md bg-primary px-3.5 py-2.5 text-center text-base font-bold text-white shadow-sm hover:bg-primary/80">{session ? "Sign out" : "Sign in"}</Link>
+				{session ? (
+					<SignOut />
+        ) : (
+            <Button
+                variant="default"
+                size="lg"
+            >
+                <Link href="/auth/sign-in">Sign in</Link>
```
Comment (phlearning) [inline src/components/header.tsx]: I can see a SignOut component and I recommend to also create a SignIn component to be consistent

**Review by phlearning — APPROVED** [review]
Looks good to me

File: src/components/header.tsx (hunk 1 above)
Comment (SabaLeonel) [reply src/components/header.tsx]: I separated the signin button into a component to make the header more readable.

**Pull request merged**